pandas/core/dtypes/cast.py

```diff
@@ -159,6 +159,31 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+# TODO: this should be a TypeVar
```

Comment:
what is this todo about?

Reply:
If the input is a Series the return type should be a Series; if the input is a Scalar the return type should be a Scalar. That's stronger than we currently check for (Series->Scalar would be allowed AFAIK)

I've had some trouble with mypy on this so want to leave for follow-on